### PR TITLE
Announcement computes intervals from last track point.

### PR DESCRIPTION
**Pull request by rgmf**
**Describe the pull request**
VoiceAnnouncement computed all intervals again and again.

In this PR VoiceAnnouncement computes intervals from last TrackPoint.

Is it enough? I'm not sure because `ContentProviderUtils.getLastValidTrackPoint` method is deprecated. Why is that method deprecated?

**Link to the the issue**
#645 

**License agreement**
By opening this pull request, you are providing your contribution under the _Apache License 2.0_ (see [LICENSE.md](LICENSE.md)).

### CODE REVIEW

**Comment by dennisguse** [comment]
getLastValidTrackPoint is deprecated as it uses a db to fetch something that is already available in memory. Day should be in TrackPointCreator ;)

**Comment by dennisguse** [comment]
getLastValidTrackPoint is deprecated as it uses a db to fetch something that is already available in memory. Day should be in TrackPointCreator ;)

File: src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncement.java [hunk 1]
```diff
@@ -183,6 +193,8 @@ public void announce(@NonNull Track track) {
 
         // We don't care about the utterance id. It is supplied here to force onUtteranceCompleted to be called.
         tts.speak(announcement, TextToSpeech.QUEUE_FLUSH, null, "not used");
+
+        startTrackPoint = contentProviderUtils.getLastValidTrackPoint(track.getId());
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncement.java:197]: startTrackPoint is the TrackPoint from the previous announcement?

What about passing it into `update()` and then pass it to `announce()`?
TrackRecordingService should have it available.

More problems with getLastValidTrackPoint: 
* what is a valid trackpoint? (it was a location previously, right?)
* it does not take care of segments
  (might be a problem if one resumes a recording, has a time-based announcement, and no new trackpoints on first announcement)

**Review by dennisguse — CHANGES_REQUESTED** [review]
Can we omit using getLastValidTrackPoint?

File: src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncement.java (hunk 1 above)
Comment (rgmf) [reply src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncement.java:197]: > What about passing it into `update()` and then pass it to `announce()`? TrackRecordingService should have it available.


That was my first thought, but I decided not to use it for looking for the easier approach. I back to my first though (I thought it was a worse approach :D).


> 
>     * it does not take care of segments
>       (might be a problem if one resumes a recording, has a time-based announcement, and no new trackpoints on first announcement)


IMHO interval is a segment between two points: A and B, and every interval shows the time (and stats)   takes the user to travel from A to B. So, in that approach pauses, stops... don't take account.

The more you pause/stop, the more it takes you to go from A to B.

I was thinking on that when I implemented intervals at the begining.

**Comment by dennisguse** [comment]
One more thing: accessing only stored data might have some pitfalls.
For example, somebody using voice announcement 100m with min recording distance of 1000m.
Then the voice announcements would rarely be correct.

**Comment by rgmf** [comment]
> One more thing: accessing only stored data might have some pitfalls. For example, somebody using voice announcement 100m with min recording distance of 1000m. Then the voice announcements would rarely be correct.

You're right. That's a good point I overlooked, thank you very much ;)

**Comment by dennisguse** [comment]
Is this ready?

**Comment by rgmf** [comment]
@dennisguse It's ready... By the way, I fixed a just discovered bug.

**Comment by dennisguse** [comment]
@rgmf can you rebase onto main?

**Pull request merged**